### ISSUE TITLE
[WIP] Train_test_split assist - improvement D - diagnostics on drift

### ISSUE THREAD

**Issue by MarieS-WiMLDS** [probabl-ai/skore#667]
Improvement of : https://github.com/probabl-ai/skore/issues/492

Former warning 7 from train_test_split skore function:  
- output: (the goal is to know is there is a drift between train & test) 
  - a [Kolmogorov-Smirnov test](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html) on continuous variables. If pvalue < 0.05, flag the feature as drifted
  - a [chi2 contingency test](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html) on categorical variables. If pvalue < 0.05, flag the feature as drifted. 
  - After the split, train a pipeline {OneHotEncorder + RandomForest} on the continuous features not flagged as drifted by Spearman + the categorical ones on the target {is_train_set}, with max_depth = 4. On this model, do a [permutation_test_score](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.permutation_test_score.html#sklearn.model_selection.permutation_test_score). If the p-value < 0.05, then display "Based on permutation, there is no drift" (cf [this example](https://scikit-learn.org/stable/auto_examples/model_selection/plot_permutation_tests_for_classification.html#sphx-glr-auto-examples-model-selection-plot-permutation-tests-for-classification-py) in sklearn doc. Otherwise, display "there is drift", and the [feature importance](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier.feature_importances_). If there is more than 100 000 points, undersample to avoid computation too long. 
- Trigger:
  - The `shuffle` parameter is `False`.

But actually, more than a warning, it's could be designed as a diagnostic, because it adds outputs.

**Comment by MarieS-WiMLDS** [comment]
close for the same reason as: https://github.com/probabl-ai/skore/issues/550, not interesting for now, much more MLOps than what we are doing + the data scientists interested in this use other tools, such as feature store.
